src/py/flwr/client/node_state.py

```diff
@@ -48,11 +48,23 @@ def register_context(
         self,
         run_id: int,
         run: Optional[Run] = None,
-        flwr_dir: Optional[Path] = None,
+        flwr_path: Optional[Path] = None,
+        app_dir: Optional[str] = None,
     ) -> None:
         """Register new run context for this node."""
         if run_id not in self.run_infos:
-            initial_run_config = get_fused_config(run, flwr_dir) if run else {}
+            initial_run_config = {}
+            if app_dir:
+                # Load from FAB-like directory
+                app_path = Path(app_dir)
+                if app_path.is_dir():
```

Comment:
Should we raise an error if this is `False`?

Reply:
yes. done